### PR TITLE
test: 1454 refactor modal pages batch 4

### PR DESCRIPTION
**Pull request by SamuelSalas**
## **Description**

We are aiming to refactor the page objects in the modal folder so that they strictly follow the page object model pattern. This would aide in providing more readable and help standardize the way we create our tests. Because of the amount of files inside of the modal folder, the scope was divided into a batch of 4.

Files inside the scope for this batch:

SigningModal.js
SkipAccountSecurityModal.js
TermsOfUseModal.js
WalletActionsModal.js
WhatsNewModal.js

## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Review by brianacnguyen — DISMISSED** [review]
Checked in with @cortisiko regarding testID changes. approved

**Review by cortisiko — CHANGES_REQUESTED** [review]
Before I review this, can you post test results from appium? thanks

**Comment by SamuelSalas** [comment]
@cortisiko Appium test run:
/wdio/features/Performance/WarmStartPerformance.feature: Failed because it was not able to find element ("//*[@text='CONNECT']"). Most likely a scroll is missing
/wdio/features/Performance/ColdStartLaunchTimes.feature: 
[0-24] The Launch time is: 16274
[0-24] Error in "1: Then the app should launch within "14" seconds"
Error: expect(received).toBeLessThan(expected)
Expected: < 14000
Received:   16274

**Comment by SamuelSalas** [comment]
Detox test: https://app.bitrise.io/app/be69d4368ee7e86d/pipelines/e36a4091-c27e-4e04-ada8-e5b0704c1711

File: e2e/viewHelper.js [hunk 1]
```diff
@@ -38,12 +38,12 @@ const validAccount = Accounts.getValidAccount();
 export const acceptTermOfUse = async () => {
   // tap on accept term of use screen
   await TestHelpers.delay(3500);
-  await TermsOfUseModal.isDisplayed();
```
Comment (Cal-L) [inline e2e/viewHelper.js:41]: Is this style of Assertion recommended moving forward?

**Comment by Cal-L** [comment]
> Detox test: https://app.bitrise.io/app/be69d4368ee7e86d/pipelines/e36a4091-c27e-4e04-ada8-e5b0704c1711

Tests are failing

**Comment by SamuelSalas** [comment]
Regression test passed: https://app.bitrise.io/app/be69d4368ee7e86d/pipelines/3f1ad255-2273-4fe2-930c-1644bbdf3a40
Smoke test passed: https://app.bitrise.io/app/be69d4368ee7e86d/pipelines/6dca24ea-3893-4678-8919-f02d7707ee7e
Performance test failed: https://app.bitrise.io/build/3d494380-28cd-48d5-a547-c5899d417db7

Cold performance test because the actual time is bigger than the expected time. I'll create an issue where it can be investigated why the increase of this time.
Warm performance test is failing because of some steps are missing. I'll create an issue where this can be resolved.

cc: @chrisleewilcox @cortisiko @Cal-L

File: e2e/viewHelper.js (hunk 1 above)
Comment (SamuelSalas) [reply e2e/viewHelper.js:41]: Yes, we have been using this implementation so far

Comment (SamuelSalas) [reply e2e/viewHelper.js:41]: If there's a better approach, i think the team would have any issue to integrate it.

**Review by chrisleewilcox — APPROVED** [review]
LGTM

File: e2e/viewHelper.js (hunk 1 above)
Comment (Cal-L) [reply e2e/viewHelper.js:41]: Sounds good

**Comment by Cal-L** [comment]
> Cold performance test because the actual time is bigger than the expected time. I'll create an issue where it can be investigated why the increase of this time.
> Warm performance test is failing because of some steps are missing. I'll create an issue where this can be resolved.

@SamuelSalas  Could  you provide some insight on why the refactor would result in slower tests? This tells me that we should not rely on these performance tests since the times are easily affected by test changes and not actual loading times.

**Review by cortisiko — APPROVED** [review]
✅ ✅

**Pull request merged**